src/ros.h

```diff
@@ -108,6 +109,57 @@ Q_OBJECT
 };
 
 /**
+ * @brief A QtQuick item that follows a ROS String published on a topic 'topic'.
+ */
+class RosStringSubscriber : public QQuickItem {
+Q_OBJECT
+    Q_PROPERTY(QString message MEMBER _message NOTIFY onMessageChanged)
+    Q_PROPERTY(QString topic WRITE setTopic MEMBER _topic)
+
+public:
+
+    RosStringSubscriber(QQuickItem* parent = 0){}
+
+    virtual ~RosStringSubscriber() {}
+
+    void setTopic(QString topic);
+    void onIncomingString(const std_msgs::String& str);
+
+signals:
+    void onMessageChanged();
+
+private:
+
+    QString _topic;
+    QString _message;
+
+    ros::NodeHandle _node;
+    ros::Subscriber _incoming_message;
+};
+
+/**
+ * @brief A QtQuick item that publish a ROS string on a topic 'topic'.
+ *
+ */
+class RosStringPublisher : public QQuickItem {
+Q_OBJECT
+    Q_PROPERTY(QString topic WRITE setTopic MEMBER _topic)
+
+public:
+    RosStringPublisher(QQuickItem* parent = 0){}
+    virtual ~RosStringPublisher() {}
+
+    void setTopic(QString topic);
+    Q_INVOKABLE void publish(QString str);
```

Comment:
We might want to keep a `publish` method to explicitly publish the message, but I would also publish everytime the text is changed with a setter: `Q_PROPERTY(QString text WRITE setText MEMBER _text)`